packages/debug/src/browser/editor/debug-inline-value-decorator.ts

```diff
@@ -240,3 +374,13 @@ export class DebugInlineValueDecorator implements FrontendApplicationContributio
     }
 
 }
+/**
+ * @returns New array with all falsy values removed. The original array IS NOT modified.
+ */
+export function coalesce<T>(array: ReadonlyArray<T | undefined | null>): T[] {
+    return <T[]>array.filter(e => !!e);
+}
+
+export function flatten<T>(arr: T[][]): T[] {
+    return (<T[]>[]).concat(...arr);
+}
```

Comment:
`coalesce` is already available in the types utility file:

https://github.com/eclipse-theia/theia/blob/d3f6359934d025781810c467e2f3b42b55f0ff0d/packages/core/src/common/types.ts#L166-L171

and `flatten` should also be added there. Or, since `debug` depends on `monaco-editor-core`, they could be imported from there.

Reply:
Thanks for catching that, I forgot to remove those two functions again, as they are not needed with the current state anymore.